src/main/java/uk/ac/sanger/sccp/stan/service/PermServiceImp.java

```diff
@@ -200,27 +255,74 @@ public void validatePermValues(Collection<String> problems, Collection<PermData>
         if (!addressesNeither.isEmpty()) {
             problems.add("Neither control type nor time specified for the given address: "+addressesNeither);
         }
+        if (!addressesWithBarcodeNotPosCon.isEmpty()) {
+            problems.add("Control barcode specified for address that is not a positive control: "+addressesWithBarcodeNotPosCon);
+        }
     }
 
     /**
      * Records the requested operation and measurements, after validation.
      * @param user the user responsible for the operation
      * @param lw the labware involved in the operation
      * @param permData the perm data to record as measurements
+     * @param controlLabware the labware specified for positive controls, if any
      * @param work the work to associate the operation with, or null
      * @return the operation recorded and the labware
      */
-    public OperationResult record(User user, Labware lw, Collection<PermData> permData, Work work) {
+    public OperationResult record(User user, Labware lw, Collection<PermData> permData, Labware controlLabware, Work work) {
         OperationType opType = opTypeRepo.getByName("Visium permabilisation");
+
```

Comment:
Spelling - 'Visium perm**e**abilisation' change required here?

Reply:
It will need to be changed, but not until it is changed everywhere. If it was spelled right here, it would not find the operation type in the database.